core/src/main/java/io/questdb/log/LogRollingFileWriter.java

```diff
@@ -324,6 +368,52 @@ private void pushFileStackUp() {
         }
     }
 
+    private void removeExcessiveLogsVisitor(long filePointer, int type) {
+        path.trimTo(logDir.length());
+        path.concat(filePointer).$();
+        if (type == Files.DT_FILE) {
+            sortedFiles.put(ff.getLastModified(path.$()), filePointer);
```

Comment:
File pointer is reused between `ff.iterateDir()` iterations. So, while the pointer stays the same, the memory contains a different null-terminated string. We need to copy the path somewhere, e.g. a reusable `StringSink` or a `String`.